store/tikv/coprocessor.go

```diff
@@ -231,28 +267,45 @@ func buildCopTasks(bo *Backoffer, cache *RegionCache, ranges *copRanges, req *kv
 		tableStart, tableEnd = keyRange[0].StartKey, keyRange[0].EndKey
 	}
 
+	if shouldStop(finishCh) {
```

Comment:
We should not only check `finishCh` here, but also check it each time a task sends to receiver, to make sure it will not be blocked by receiver channel if `copIteratorTaskSender.run` exits.